src/main/kotlin/org/rust/ide/utils/checkMatch/Pattern.kt

```diff
@@ -23,10 +21,10 @@ data class Pattern(val ty: Ty, val kind: PatternKind) {
             is PatternKind.Binding -> kind.name
 
             is PatternKind.Variant -> {
-                val variantName = kind.variant.name.orEmpty()
-                val itemName = kind.item.name
+                val variantName = kind.variant.name.orEmpty().escapeIdentifierIfNeeded()
+                val itemName = kind.item.name.orEmpty().escapeIdentifierIfNeeded()
 
-                val variantInScope = ctx?.findInScope(variantName, VALUES)
+                val variantInScope = ctx?.findInScope(variantName.unescapeIdentifier(), VALUES)
```

Comment:
Let's avoid unnecessary escaping and unescaping.
I suggest call `.escapeIdentifierIfNeeded()` in the following code below
```kotlin
val name = if (variantInScope != null) {
    variantName
} else {
    "$itemName::$variantName"
}
```

Note, it will require additional tests when all enum variants are imported. Something like
```rust
enum r#type { A, r#if, r#fn }

fn main() {
    use r#type::*;
    
    let a = A;
    match a/*caret*/ {

    }
}
```